lib/_tls_wrap.js

```diff
@@ -464,6 +464,7 @@ function TLSSocket(socket, opts) {
   this._securePending = false;
   this._newSessionPending = false;
   this._controlReleased = false;
+  this.secureConnecting = true;
```

Comment:
This is never set to `false` on client sockets.
So if `Http2Session` constructor (in `lib/internal/http2/core.js`) is called _after_ `secureConnect` is emitted, then `secureConnecting` is still `true` but it'll end up never calling `setupHandle` and `connect` will never be emitted.